extensions/amp-analytics/0.1/crypto-impl.js

```diff
@@ -38,27 +47,31 @@ export class Crypto {
    * @throws {!Error} when input string contains chars out of range [0,255]
    */
   sha384(input) {
-    if (this.subtle_) {
-      try {
-        return this.subtle_.digest({name: 'SHA-384'},
-                input instanceof Uint8Array ? input : stringToBytes(input))
-            // [].slice.call(Unit8Array) is a shim for Array.from(Unit8Array)
-            /** @param {?} buffer */
-            .then(buffer => [].slice.call(new Uint8Array(buffer)),
-                e => {
-                  // Chrome doesn't allow the usage of Crypto API under
-                  // non-secure origin: https://www.chromium.org/Home/chromium-security/prefer-secure-origins-for-powerful-new-features
-                  if (e.message && e.message.indexOf('secure origin') < 0) {
-                    // Log unexpected fallback.
-                    dev().error(TAG, FALLBACK_MSG, e);
-                  }
-                  return lib.sha384(input);
-                });
-      } catch (e) {
-        dev().error(TAG, FALLBACK_MSG, e);
-      }
+    // polyfill is (being) loaded,
+    // means native Crypto API is not available or failed before.
+    if (this.polyfillPromise_) {
+      return this.polyfillPromise_.then(polyfill => polyfill.sha384(input));
+    }
+    try {
+      return this.subtle_.digest({name: 'SHA-384'},
+              input instanceof Uint8Array ? input : stringToBytes(input))
+          // [].slice.call(Unit8Array) is a shim for Array.from(Unit8Array)
+          /** @param {?} buffer */
+          .then(buffer => [].slice.call(new Uint8Array(buffer)),
+              e => {
+                // Chrome doesn't allow the usage of Crypto API under
+                // non-secure origin: https://www.chromium.org/Home/chromium-security/prefer-secure-origins-for-powerful-new-features
+                if (e.message && e.message.indexOf('secure origin') < 0) {
+                  // Log unexpected fallback.
+                  dev().error(TAG, FALLBACK_MSG, e);
+                }
+                return this.loadPolyfill_()
+                    .then(polyfill => polyfill.sha384(input));
```

Comment:
that will not work. we need to call loadPolyfill here otherwise this.polyfillPromise_ is null forever